{
  "active_lock_reason": null,
  "assignee": {
    "avatar_url": "https://avatars0.githubusercontent.com/u/278?v=4",
    "events_url": "https://api.github.com/users/gnufied/events{/privacy}",
    "followers_url": "https://api.github.com/users/gnufied/followers",
    "following_url": "https://api.github.com/users/gnufied/following{/other_user}",
    "gists_url": "https://api.github.com/users/gnufied/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/gnufied",
    "id": 278,
    "login": "gnufied",
    "node_id": "MDQ6VXNlcjI3OA==",
    "organizations_url": "https://api.github.com/users/gnufied/orgs",
    "received_events_url": "https://api.github.com/users/gnufied/received_events",
    "repos_url": "https://api.github.com/users/gnufied/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/gnufied/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/gnufied/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/gnufied"
  },
  "assignees": [
    {
      "avatar_url": "https://avatars0.githubusercontent.com/u/278?v=4",
      "events_url": "https://api.github.com/users/gnufied/events{/privacy}",
      "followers_url": "https://api.github.com/users/gnufied/followers",
      "following_url": "https://api.github.com/users/gnufied/following{/other_user}",
      "gists_url": "https://api.github.com/users/gnufied/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/gnufied",
      "id": 278,
      "login": "gnufied",
      "node_id": "MDQ6VXNlcjI3OA==",
      "organizations_url": "https://api.github.com/users/gnufied/orgs",
      "received_events_url": "https://api.github.com/users/gnufied/received_events",
      "repos_url": "https://api.github.com/users/gnufied/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/gnufied/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gnufied/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/gnufied"
    },
    {
      "avatar_url": "https://avatars1.githubusercontent.com/u/22076?v=4",
      "events_url": "https://api.github.com/users/pohly/events{/privacy}",
      "followers_url": "https://api.github.com/users/pohly/followers",
      "following_url": "https://api.github.com/users/pohly/following{/other_user}",
      "gists_url": "https://api.github.com/users/pohly/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/pohly",
      "id": 22076,
      "login": "pohly",
      "node_id": "MDQ6VXNlcjIyMDc2",
      "organizations_url": "https://api.github.com/users/pohly/orgs",
      "received_events_url": "https://api.github.com/users/pohly/received_events",
      "repos_url": "https://api.github.com/users/pohly/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/pohly/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/pohly/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/pohly"
    },
    {
      "avatar_url": "https://avatars1.githubusercontent.com/u/10052848?v=4",
      "events_url": "https://api.github.com/users/saad-ali/events{/privacy}",
      "followers_url": "https://api.github.com/users/saad-ali/followers",
      "following_url": "https://api.github.com/users/saad-ali/following{/other_user}",
      "gists_url": "https://api.github.com/users/saad-ali/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/saad-ali",
      "id": 10052848,
      "login": "saad-ali",
      "node_id": "MDQ6VXNlcjEwMDUyODQ4",
      "organizations_url": "https://api.github.com/users/saad-ali/orgs",
      "received_events_url": "https://api.github.com/users/saad-ali/received_events",
      "repos_url": "https://api.github.com/users/saad-ali/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/saad-ali/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/saad-ali/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/saad-ali"
    },
    {
      "avatar_url": "https://avatars3.githubusercontent.com/u/24448061?v=4",
      "events_url": "https://api.github.com/users/msau42/events{/privacy}",
      "followers_url": "https://api.github.com/users/msau42/followers",
      "following_url": "https://api.github.com/users/msau42/following{/other_user}",
      "gists_url": "https://api.github.com/users/msau42/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/msau42",
      "id": 24448061,
      "login": "msau42",
      "node_id": "MDQ6VXNlcjI0NDQ4MDYx",
      "organizations_url": "https://api.github.com/users/msau42/orgs",
      "received_events_url": "https://api.github.com/users/msau42/received_events",
      "repos_url": "https://api.github.com/users/msau42/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/msau42/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/msau42/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/msau42"
    }
  ],
  "author_association": "CONTRIBUTOR",
  "body": "The test `[sig-storage] CSI Volumes CSI plugin test using CSI driver: hostPath should provision storage` flakes very occasionally. Below are the findings of the circumstances of the flake.\r\n\r\nDuring `testDynamicProvisioning` two pods are created in quick succession. The test flakes when the second Pod fails to come up.\r\n\r\nThe second pod fails to come up with error:\r\n`MountVolume.WaitForAttach failed for volume \"pvc-bc2294b873ef11e8\" : resource name may not be empty`\r\n\r\nUpon exploring the logs we discovered that there was only 1 Attach call made at the beginning, and one Detach call upon timeout of the test [See timestamps]:\r\n```\r\nI0613 19:00:41.888473       1 reconciler.go:291] attacherDetacher.AttachVolume started for volume \"pvc-0c896e426f3c11e8\" (UniqueName: \"kubernetes.io/csi/csi-hostpath^0da20430-6f3c-11e8-a80a-0a580a40016c\") from node \"e2e-40645-674b9-minion-group-gz10\" \r\nI0613 19:00:42.533343       1 operation_generator.go:335] AttachVolume.Attach succeeded for volume \"pvc-0c896e426f3c11e8\" (UniqueName: \"kubernetes.io/csi/csi-hostpath^0da20430-6f3c-11e8-a80a-0a580a40016c\") from node \"e2e-40645-674b9-minion-group-gz10\" \r\nI0613 19:15:52.528902       1 reconciler.go:233] attacherDetacher.DetachVolume started for volume \"pvc-0c896e426f3c11e8\" (UniqueName: \"kubernetes.io/csi/csi-hostpath^0da20430-6f3c-11e8-a80a-0a580a40016c\") on node \"e2e-40645-674b9-minion-group-gz10\" \r\nI0613 19:15:52.615262       1 operation_generator.go:1225] Verified volume is safe to detach for volume \"pvc-0c896e426f3c11e8\" (UniqueName: \"kubernetes.io/csi/csi-hostpath^0da20430-6f3c-11e8-a80a-0a580a40016c\") on node \"e2e-40645-674b9-minion-group-gz10\" \r\nI0613 19:15:53.367487       1 operation_generator.go:419] DetachVolume.Detach succeeded for volume \"pvc-0c896e426f3c11e8\" (UniqueName: \"kubernetes.io/csi/csi-hostpath^0da20430-6f3c-11e8-a80a-0a580a40016c\") on node \"e2e-40645-674b9-minion-group-gz10\" \r\n```\r\n\r\nWe tracked down the `resource name may not be empty` error to the likely culprit in func `waitForVolumeAttachmentInternal`:\r\n```\r\nattach, err := c.k8s.StorageV1beta1().VolumeAttachments().Get(attachID, meta.GetOptions{})\r\n```\r\nbased on the error message returned. \r\n\r\nThis is likely caused by `attachID`(string) being empty [uncomfirmed], which is what the Kubelet would get from the DevicePath field on the Node object. This DevicePath field is normally added on Attach and removed on Detach of the volume.\r\n\r\nThe DevicePath is clearly present on the Node when the first pod is being brought up. But it is possible that it is somehow missing or not able to be retrieved by the time the second Pod is starting up (and continues to be unavailable exhausting the 15m timeout on the test) even though no Detach operation was performed on the volume in the meantime (according to logs). We are currently unable to reproduce this flake on command and did all analysis based on logs from the two flakes we saw naturally occur in the 100s of runs performed of the test.\r\n\r\nThis flake was originally seen in #65064 and the test logs can be found here: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/65064/pull-kubernetes-e2e-gce/40645/\r\n\r\n**Environment**: GCE\r\n- Kubernetes version (use `kubectl version`):\r\n```\r\nClient Version: version.Info{Major:\"1\", Minor:\"8\", GitVersion:\"v1.8.6\", GitCommit:\"6260bb08c46c31eea6cb538b34a9ceb3e406689c\", GitTreeState:\"clean\", BuildDate:\"2017-12-21T06:34:11Z\", GoVersion:\"go1.8.3\", Compiler:\"gc\", Platform:\"linux/amd64\"}\r\nServer Version: version.Info{Major:\"1\", Minor:\"12+\", GitVersion:\"v1.12.0-alpha.0.975+98194b8630958e\", GitCommit:\"98194b8630958ee48db662f42404dc0bc6a2ed80\", GitTreeState:\"clean\", BuildDate:\"2018-06-19T18:16:46Z\", GoVersion:\"go1.10.2\", Compiler:\"gc\", Platform:\"linux/amd64\"}\r\n```\r\n- Cloud provider or hardware configuration: GCP\r\n- OS (e.g. from /etc/os-release):\r\n```\r\nBUILD_ID=10323.64.0\r\nNAME=\"Container-Optimized OS\"\r\nKERNEL_COMMIT_ID=94bdaa2af49562e7aba0168cae050911d105f5cb\r\nGOOGLE_CRASH_ID=Lakitu\r\nVERSION_ID=65\r\nBUG_REPORT_URL=https://crbug.com/new\r\nPRETTY_NAME=\"Container-Optimized OS from Google\"\r\nVERSION=65\r\nGOOGLE_METRICS_PRODUCT_ID=26\r\nHOME_URL=\"https://cloud.google.com/compute/docs/containers/vm-image/\"\r\nID=cos\r\n```\r\n- Kernel (e.g. `uname -a`): `Linux e2e-test-dyzz-master 4.4.111+ #1 SMP Wed Mar 28 13:44:38 PDT 2018 x86_64 Intel(R) Xeon(R) CPU @ 2.30GHz GenuineIntel GNU/Linux`\r\n\r\n/kind flake\r\n/kind bug\r\n/sig storage\r\n/cc @msau42 @jingxu97 @saad-ali @verult ",
  "closed_at": "2018-11-19T19:33:19Z",
  "closed_by": {
    "avatar_url": "https://avatars0.githubusercontent.com/u/20407524?v=4",
    "events_url": "https://api.github.com/users/k8s-ci-robot/events{/privacy}",
    "followers_url": "https://api.github.com/users/k8s-ci-robot/followers",
    "following_url": "https://api.github.com/users/k8s-ci-robot/following{/other_user}",
    "gists_url": "https://api.github.com/users/k8s-ci-robot/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/k8s-ci-robot",
    "id": 20407524,
    "login": "k8s-ci-robot",
    "node_id": "MDQ6VXNlcjIwNDA3NTI0",
    "organizations_url": "https://api.github.com/users/k8s-ci-robot/orgs",
    "received_events_url": "https://api.github.com/users/k8s-ci-robot/received_events",
    "repos_url": "https://api.github.com/users/k8s-ci-robot/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/k8s-ci-robot/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/k8s-ci-robot/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/k8s-ci-robot"
  },
  "comments": 45,
  "comments_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/65246/comments",
  "created_at": "2018-06-19T22:22:46Z",
  "events_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/65246/events",
  "html_url": "https://github.com/kubernetes/kubernetes/issues/65246",
  "id": 333860191,
  "labels": [
    {
      "color": "e11d21",
      "default": false,
      "description": "Categorizes issue or PR as related to a bug.",
      "id": 105146071,
      "name": "kind/bug",
      "node_id": "MDU6TGFiZWwxMDUxNDYwNzE=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/kind/bug"
    },
    {
      "color": "e11d21",
      "default": false,
      "description": "Categorizes issue or PR as related to a consistently or frequently failing test.",
      "id": 496752468,
      "name": "kind/failing-test",
      "node_id": "MDU6TGFiZWw0OTY3NTI0Njg=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/kind/failing-test"
    },
    {
      "color": "f7c6c7",
      "default": false,
      "description": "Categorizes issue or PR as related to a flaky test.",
      "id": 264749912,
      "name": "kind/flake",
      "node_id": "MDU6TGFiZWwyNjQ3NDk5MTI=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/kind/flake"
    },
    {
      "color": "e11d21",
      "default": false,
      "description": "Highest priority. Must be actively worked on as someone's top priority right now.",
      "id": 114528068,
      "name": "priority/critical-urgent",
      "node_id": "MDU6TGFiZWwxMTQ1MjgwNjg=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/priority/critical-urgent"
    },
    {
      "color": "d2b48c",
      "default": false,
      "description": "Categorizes an issue or PR as relevant to SIG Storage.",
      "id": 169428334,
      "name": "sig/storage",
      "node_id": "MDU6TGFiZWwxNjk0MjgzMzQ=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/sig/storage"
    }
  ],
  "labels_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/65246/labels{/name}",
  "locked": false,
  "milestone": {
    "closed_at": "2020-08-23T05:01:14Z",
    "closed_issues": 462,
    "created_at": "2017-11-15T06:32:59Z",
    "creator": {
      "avatar_url": "https://avatars0.githubusercontent.com/u/5595220?v=4",
      "events_url": "https://api.github.com/users/thockin/events{/privacy}",
      "followers_url": "https://api.github.com/users/thockin/followers",
      "following_url": "https://api.github.com/users/thockin/following{/other_user}",
      "gists_url": "https://api.github.com/users/thockin/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/thockin",
      "id": 5595220,
      "login": "thockin",
      "node_id": "MDQ6VXNlcjU1OTUyMjA=",
      "organizations_url": "https://api.github.com/users/thockin/orgs",
      "received_events_url": "https://api.github.com/users/thockin/received_events",
      "repos_url": "https://api.github.com/users/thockin/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/thockin/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/thockin/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/thockin"
    },
    "description": "",
    "due_on": null,
    "html_url": "https://github.com/kubernetes/kubernetes/milestone/40",
    "id": 2913713,
    "labels_url": "https://api.github.com/repos/kubernetes/kubernetes/milestones/40/labels",
    "node_id": "MDk6TWlsZXN0b25lMjkxMzcxMw==",
    "number": 40,
    "open_issues": 0,
    "state": "closed",
    "title": "v1.13",
    "updated_at": "2020-08-23T05:01:14Z",
    "url": "https://api.github.com/repos/kubernetes/kubernetes/milestones/40"
  },
  "node_id": "MDU6SXNzdWUzMzM4NjAxOTE=",
  "number": 65246,
  "performed_via_github_app": null,
  "repository_url": "https://api.github.com/repos/kubernetes/kubernetes",
  "state": "closed",
  "title": "Second Pod Using Same PVC Fails WaitForAttach Flakily",
  "updated_at": "2018-11-19T19:34:20Z",
  "url": "https://api.github.com/repos/kubernetes/kubernetes/issues/65246",
  "user": {
    "avatar_url": "https://avatars3.githubusercontent.com/u/5459214?v=4",
    "events_url": "https://api.github.com/users/davidz627/events{/privacy}",
    "followers_url": "https://api.github.com/users/davidz627/followers",
    "following_url": "https://api.github.com/users/davidz627/following{/other_user}",
    "gists_url": "https://api.github.com/users/davidz627/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/davidz627",
    "id": 5459214,
    "login": "davidz627",
    "node_id": "MDQ6VXNlcjU0NTkyMTQ=",
    "organizations_url": "https://api.github.com/users/davidz627/orgs",
    "received_events_url": "https://api.github.com/users/davidz627/received_events",
    "repos_url": "https://api.github.com/users/davidz627/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/davidz627/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/davidz627/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/davidz627"
  }
}